### ISSUE TITLE
Single click agent

### ISSUE THREAD

**Issue by OutbackMatt** [Ylianst/MeshCentral#790]
I have a group called 'remote'.
I have created an agent for that group, that I show on my website as a small download.

Anyone can download it and run it, and they get the standard Agent, choose to either install permanently or run temporarily

Even though there is ONLY four buttons on this agent, I think it is THREE TO MANY
I'd like a 'light' agent that ONLY does one up run temporarily, with the click of a SINGLE BUTTON once someone downloads the agent. (I'm thinking support for not very savy computer users)

Then if I needed a full install, that I could run an installer once I have connected to the client computer, and then change groups assign users etc as normal. 

I'm thinking like the 'UVNC Single Click'. 
Download, run, agree and press OK.

Easy enough for my elderly parents to do

**Comment by PathfinderNetworks** [comment]
I like this idea.  Just a one-time run client with no other options.  For me, I'd create a special device group just for this agent (so I would know where these one-time devices are going to show up in Mesh and would only be for those types of connections).

**Comment by asasin114** [comment]
This would be awesome!

**Comment by Ylianst** [comment]
There is already a way to control what buttons are made available. When you check "Add Agent" you can select the installation type: Background, Interactive or Both. If you select interactive, the install/uninstall buttons will not be present, if background only, the "Connect" button will not be present. Can you give this a try and let me know what you think? Thanks.

![x](https://user-images.githubusercontent.com/1319013/71852732-4c312580-308e-11ea-8cea-16d547355768.png)

**Comment by OutbackMatt** [comment]
That does help thanks, but it is still too wordy and talks about an install or uninstall button and mesh agents. Way too complex.

![image](https://user-images.githubusercontent.com/7231493/71869206-ed86ad00-315c-11ea-85d1-374b101e91e8.png)

I'd like
1. a custom name shown instead of 'Mesh Agent v2' (unsure if this is possible...)
2. and 3. Simple text that says something like 'Allow [MESHNAME] - [Username]' to connect to this computer remotely
4. A button that says 'ALLOW REMOTE ACCESS'
![image](https://user-images.githubusercontent.com/7231493/71869337-73a2f380-315d-11ea-8a42-5d84aee8c3a0.png)

**Comment by MailYouLater** [comment]
The "Installation Information" section could be helpful for debugging in cases where the agent doesn't connect to the server properly, but perhaps it could be in a 'Details' or 'More Info' section that's hidden by default, similar to PDFTKBuilder's "Document Protection" options.

| PDFTKBuilder with Document Protection options hidden | PDFTKBuilder with Document Protection options visible |
| - | - |
| ![PDFTKBuilder with Document Protection options hidden](https://user-images.githubusercontent.com/1279475/71914844-59522100-312f-11ea-8886-d293aa5b603e.png) | ![PDFTKBuilder with Document Protection options visible](https://user-images.githubusercontent.com/1279475/71914885-6d961e00-312f-11ea-8202-2ebabd56006f.png) |

**Comment by agottschling** [comment]
I'd really love to see this, possibly with a one time code mechanisim, kind of like teamviewer. Dunno if this is even possible with how MeshCentral is currently written though.

**Comment by MailYouLater** [comment]
@agottschling: I'm not sure I get how a One Time Password(or Code) would work with this.  Perhaps you should write up a separate proposal and post that as a [new issue](https://github.com/Ylianst/MeshCentral/issues/new), separate from this one.

**Comment by agottschling** [comment]
For a one-click app, You'd probably want some kind of security. Otherwise, you could run this on someone's machine, and get remote control without needing admin rights on the remote system. 
You may be right though, this may need to go into a separate request. Just let me know.

**Comment by MailYouLater** [comment]
This request is about simplifying and possibly customizing the UI for the installer.  I think requiring a One Time Password to connect should be considered separately from this.

**Comment by OutbackMatt** [comment]
Don't we have a one click app anyway (albeit a much too complex one)
I could simply download the meshagent from my website (2.8Mb) and install.
(In fairness I find that I need to 'run as administrator' to be more secure and stable, but it still works initially as a normal user)

It takes seconds (I've timed it) and it can be set to not leave a trace on the remote machine (the one being watched or controlled), no task bar icon, no connection bar, no user interaction.

It's actually pretty scary how quickly and easily this Mesh system is to implement and use, and then the back door command prompt and file uploads and downloads... and then the remote scripting add-in...

and all on normal web server ports, firewalls wouldn't detect it, and an admin would need to know to look for the mesh agent service.

**Comment by agottschling** [comment]
Good Point @MailYouLater,
I'll make a separate request for an OTP, I've updated my previous posts.
